src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -324,6 +325,8 @@ private async Task<WorkUnitResult> ExecuteTask(TaskExecutionMode mode, Lookup lo
 
                 WorkUnitResult aggregateResult = new WorkUnitResult();
 
+                // Some tests do not provide an actual taskNode; checking if _taskNode == null prevents those tests from failing.
+                MSBuildEventSource.Log.ExecuteTaskStart(_taskNode == null ? null : _taskNode.Name);
```

Comment:
What if the task calls IBuildEngineX.Yield? The events will bracket more than the time the task spent executing.

Reply:
@danmosemsft What would be the cleanest way of subtracting yields?